microsoft/testsuites/power/power.py

```
@@ -257,6 +259,191 @@ def verify_hibernation_max_data_disks(self, node: Node, log: Logger) -> None:
+                log_filename = log_file_path.split("/")[-1]
+                log.info(f"=== Contents of {log_filename} ===")
+
+                try:
+                    content = cat.read(log_file_path, sudo=True)
+                    log.info(f"{log_filename} content:\n{content}")
+                except Exception as cat_ex:
+                    log.debug(f"Failed to read {log_file_path}: {cat_ex}")
+
+                log.info(f"=== End of {log_filename} ===")
+
+        except Exception as log_ex:
+            log.info(f"Failed to collect extension logs: {log_ex}")
+
+    def _check_os_disk_space(self, node: Node, log: Logger) -> None:
+        try:
+            df = node.tools[Df]
+            root_partition = df.get_partition_by_mountpoint("/")
+            if root_partition:
+                available_gb = (
+                    root_partition.available_blocks / 1024 / 1024
+                )  # Convert from KB to GB
+                used_percent = root_partition.percentage_blocks_used
+                total_gb = (
+                    root_partition.total_blocks / 1024 / 1024
+                )  # Convert from KB to GB
+
+                log.info(
+                    f"OS Disk Space (/) - Total: {total_gb:.2f}GB, "
+                    f"Used: {used_percent}%, Available: {available_gb:.2f}GB"
+                )
+                # Check if low disk space might be causing issues
+                if available_gb < 1.0:  # Less than 1GB available
+                    log.info(
+                        f"LOW DISK SPACE WARNING: Only {available_gb:.2f}GB "
+                        f"available on OS disk"
+                    )
+                elif used_percent > 90:  # More than 90% used
+                    log.info(
+                        f"HIGH DISK USAGE WARNING: {used_percent}% of OS disk is used"
+                    )
+            else:
+                log.debug("Could not get root partition space information")
+        except Exception as ex:
```

If you're not sure what an exception means or how to handle it, don't try to catch it blindly. Let it bubble up so you can see what went wrong and figure out the best way to deal with it.

---

updated